sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -411,9 +450,173 @@ def tune(
                                 env_from=env_from if env_from else None,
                                 resources=resources_per_trial,
                             )
-                        ],
+            
+            pod_spec = client.V1PodTemplateSpec(
+                metadata=models.V1ObjectMeta(
+                    annotations={"sidecar.istio.io/inject": "false"}
+                ),
+                spec=client.V1PodSpec(
+                    restart_policy="Never",
+                    containers=[container_spec],
+                ),
+            )
+
+        # If users choose to use external models and datasets.
+        else:
+            try:
+                from kubeflow.storage_initializer.hugging_face import \
+                    HuggingFaceDatasetParams
+                from kubeflow.storage_initializer.hugging_face import \
+                    HuggingFaceModelParams
+                from kubeflow.storage_initializer.hugging_face import \
+                    HuggingFaceTrainerParams
+                from kubeflow.storage_initializer.s3 import S3DatasetParams
+                import peft
+                import transformers
+            except ImportError:
+                raise ImportError(
+                    "Tune API dependencies not installed. "
+                    + "Run: pip install -U 'kubeflow-training[huggingface]' "
+                )
+            
+            # Create PVC for the Storage Initializer.
```

Comment:
Please add TODO that this should be moved to the Controller.

Reply:
Got it! Already added.